projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -1036,8 +1046,7 @@ export class IgxComboComponent extends DisplayDensityBase implements IgxComboBas
             };
             this.onSearchInput.emit(args);
             if (args.cancel) {
-                this.searchValue = null;
-                return;
```

Comment:
I think this could probably stay, as the match check is still unnecessary in this case